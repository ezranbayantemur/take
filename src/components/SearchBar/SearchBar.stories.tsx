import type {Meta, StoryObj} from '@storybook/react-native';
import {SearchBar} from '@components';
import type {SearchBarProps} from './SearchBar.types';

const SearchBarMeta: Meta<SearchBarProps<any>> = {
  args: {
    placeholder: 'Ara...',
  },
  title: 'SearchBar',
  component: SearchBar,
};

export default SearchBarMeta;

export const Basic: StoryObj<SearchBarProps<any>> = {};
