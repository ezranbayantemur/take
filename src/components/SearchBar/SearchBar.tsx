import {View} from 'react-native';
import React from 'react';
import type {SearchBarProps} from './SearchBar.types';

import Input from '../Input';

const SearchBar = ({
  testID,
  placeholder = 'Ara...',
  debounceTime = 200,
  onSearch,
}: SearchBarProps) => {
  return (
    <View>
      <Input
        testID={`${testID}_searchbar`}
        placeholder={placeholder}
        debounceTime={debounceTime}
        onChangeText={onSearch}
      />
    </View>
  );
};

export default React.memo(SearchBar);
