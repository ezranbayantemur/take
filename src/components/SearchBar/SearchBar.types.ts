export interface SearchBarProps {
  testID: string;
  onSearch: (value: string) => void;
  placeholder?: string;
  debounceTime?: number;
}
