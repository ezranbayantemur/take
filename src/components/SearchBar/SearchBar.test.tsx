import React from 'react';
import {act, fireEvent, render} from '@testing-library/react-native';
import SearchBar from './SearchBar';
import products from '@mocks/products_for_category.json';

const mockOnSearch = jest.fn();

let wrapper: ReturnType<typeof render>;

jest.useFakeTimers();
describe('SearchBar unit tests', () => {
  beforeEach(() => {
    wrapper = render(
      <SearchBar testID="test" data={products} onSearch={mockOnSearch} />,
    );
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render default placeholder correctly', () => {
    expect(wrapper.queryByPlaceholderText('Ara...')).not.toBeNull();
  });

  it('should render custom placeholder correctly', () => {
    wrapper.rerender(
      <SearchBar
        data={products}
        testID="test"
        onSearch={mockOnSearch}
        placeholder="Ürün ara..."
      />,
    );
    expect(wrapper.queryByPlaceholderText('Ürün ara...')).not.toBeNull();
  });

  it('should trigger search correctly', () => {
    const input = wrapper.getByTestId('test_searchbar_input');
    act(() => {
      fireEvent(input, 'onChangeText', 'iphone');
      jest.advanceTimersByTime(200);
    });

    expect(mockOnSearch).toHaveBeenCalledWith([products[1], products[0]]);
  });

  it('should trigger search with custom debounce correctly', () => {
    wrapper.rerender(
      <SearchBar
        testID="test"
        data={products}
        onSearch={mockOnSearch}
        debounceTime={500}
      />,
    );

    const input = wrapper.getByTestId('test_searchbar_input');
    act(() => {
      fireEvent(input, 'onChangeText', 'iphone');
      jest.advanceTimersByTime(500);
    });

    expect(mockOnSearch).toHaveBeenCalledWith([products[1], products[0]]);
  });
});
