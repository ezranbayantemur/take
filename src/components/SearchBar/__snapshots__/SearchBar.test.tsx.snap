// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar unit tests should render correctly 1`] = `
<View
  style={{}}
>
  <View>
    <View
      style={
        {
          "backgroundColor": "#E1E1E1",
          "borderRadius": 5,
          "margin": 8,
          "padding": 16,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Ara..."
        style={
          {
            "fontFamily": "Lato-Regular",
            "fontSize": 15,
          }
        }
        testID="test_searchbar_input"
      />
    </View>
  </View>
</View>
`;
